internal/provider/resource_cluster_user.go

```diff
@@ -0,0 +1,317 @@
+	var (
+		data  ClusterUserModel
+		state ClusterUserModel
+	)
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	stateNsID, stateUsername := parseClusterUserIdentifier(state.ID.ValueString(), &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if data.Username != state.Username {
+		resp.Diagnostics.AddError(
+			"Cannot update immutable field",
+			fmt.Sprintf("Username cannot be updated, previous: %s, new: %s", state.Username, data.Username),
+		)
+		return
+	}
+
+	if data.CreateDB != state.CreateDB {
+		resp.Diagnostics.AddError(
+			"Cannot update immutable field",
+			fmt.Sprintf("CreateDB cannot be updated, previous: %s, new: %s", state.CreateDB, data.CreateDB),
+		)
+		return
+	}
+
+	if data.SuperUser != state.SuperUser {
+		resp.Diagnostics.AddError(
+			"Cannot update immutable field",
+			fmt.Sprintf("SuperUser cannot be updated, previous: %s, new: %s", state.SuperUser, data.SuperUser),
+		)
+		return
+	}
+
+	if data.Password != state.Password {
+		if err := r.client.UpdateClusterUserPassword(ctx, stateNsID, stateUsername, data.Password.ValueString()); err != nil {
+			resp.Diagnostics.AddError("Unable to update cluster user password", err.Error())
+			return
+		}
+	}
+
+	user, err := r.client.GetClusterUser(ctx, stateNsID, state.Username.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to read cluster user", err.Error())
+		return
+	}
+
+	// the password is stored in the state to avoid inconsistency error.
+	clusterUserToDataModel(stateNsID, user, &data)
+
+	// Directly save the plan to the state since we cannot know the password through API.
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *ClusterUserResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var data ClusterUserModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if data.ID.IsUnknown() || data.ID.IsNull() {
+		resp.Diagnostics.AddError("ID is missing", "ID is required to delete the resource")
+		return
+	}
+
+	nsID, username := parseClusterUserIdentifier(data.ID.ValueString(), &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if err := r.client.DeleteClusterUser(ctx, nsID, username); err != nil {
+		resp.Diagnostics.AddError("Unable to delete cluster user", err.Error())
+		return
+	}
+}
+
+func (r *ClusterUserResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	nsID, username := parseClusterUserIdentifier(req.ID, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if _, err := r.client.GetClusterUser(ctx, nsID, username); err != nil {
```

Comment:
To confirm the behavior: password won't be populated after the import and the next apply will update the password to the one specified in the config, right?

Reply:
correct.